File: examples/apps/python/msgpacketizer-py/src/msgpacketizerpy/msgpacketizer/encoder.py
import logging
from typing import Any

import msgpack

from ...packetizerpy import packetizer
from .common import to_hex


def encode(index: int, msgs: Any) -> bytes:
    logging.debug(f"encode input: index = {index}, msgs = {msgs}")
    packed = msgpack.packb(msgs, use_bin_type=True)
    logging.debug(f"packed: {to_hex(packed)}")
    encoded = packetizer.encode(index, packed)
    logging.debug(f"encoded: {to_hex(encoded)}")
    return encoded

File: examples/apps/go/msgpacketizer-go/pkg/msgpacketizer-go/encoder.go
package msgpacketizer

import (
	// TODO:
	// "github.com/hideakitai/packetizer-go"
	"packetizer-go"

	"github.com/sirupsen/logrus"

	"github.com/vmihailenco/msgpack/v5"
)

func Encode(index uint8, msgs interface{}) ([]byte, error) {
	logrus.Debugf("encode input: index=%d, msgs=%+v", index, msgs)
	packed, err := msgpack.Marshal(&msgs)
	if err != nil {
		logrus.Warnf("msgpack marshal error: %s", err)
		return []byte{}, err
	}
	logrus.Debugf("packed: %s", ToHex(packed))
	encoded := packetizer.Encode(index, packed)
	logrus.Debugf("encoded: %s", ToHex(encoded))
	return encoded, nil
}
